### ISSUE TITLE
[] method call should return nil if the key isn't present in hash 

### ISSUE THREAD

**Issue by meetme2meat** [UnderpantsGnome/config_reader-gem#2]
The MagicHash implementation is bits different from the traditional Ruby Hash implementation.

Example. 

> Ruby hash 
> {}['a'] => nil
> 
> MagicHash
> {}['a'] => undefined method 'a' ...  

I see the resultant code that does is [here](https://github.com/UnderpantsGnome/config_reader-gem/blob/master/lib/config_reader/magic_hash.rb#L19). 

I fail to understand the reason for MagicHash `.[]` the way it is in the spec.

But because of the current restriction with MagicHash `.[]` the below code breaks

```
ActiveRecord::Base.establish_connection(with_config_reader_options)
```

Since ActiveRecord do `spec[]` and other options check.

**Comment by UnderpantsGnome** [comment]
I don't see the behavior you describe, I get a key not found error, as is intended.

    >> ConfigReader::MagicHash.new['a']
    KeyError: key not found: :a

The reason for this is so that you aren't calling configuration options that don't exist, if you do it will throw an exception. In my top level config YAML I list out all possible keys in the `defaults` so you can easily see all available config settings.

To deal with the issue of passing config into `ActiveRecord::Base.establish_connection` you can do this

    ActiveRecord::Base.establish_connection(with_config_reader_options.to_h)

**Comment by meetme2meat** [comment]
Thanks for explaining and yes the error was `KeyError: key not found:` 

But on the other noted I just find the purpose of raising a `KeyError` for missing configuration a bit too harsh letting it return `nil` sound reasonable. But I love to understand the motivation here.

**Comment by UnderpantsGnome** [comment]
@meetme2meat I just released 1.0.5 that allows you to turn off the KeyError exception, the README has info.
